### PR TITLE
Docs: Simpler Examples, generate examples pages from actual examples code

### PR DESCRIPTION
**Pull request by sh-rp**
### Description
In this PR we fully deduplicate the examples. We generate the examples pages in the docs based on the code in the docs/examples folder, not vice versa.

TODO: 
- [x] Tests

### CODE REVIEW

File: docs/examples/connector_x_arrow/connector_x_arrow.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
+"""
```
Comment (sh-rp) [inline docs/examples/connector_x_arrow/connector_x_arrow.py:1]: note

**Review by burnash — APPROVED** [review]
LGTM

File: docs/examples/connector_x_arrow/connector_x_arrow.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+:::tip
+You can yield several tables if your data is large and you need to partition your load.
+:::
+
+We'll learn:
+
+- How to get arrow tables from [connector X](https://github.com/sfu-db/connector-x) and yield them.
+- That merge and incremental loads work with arrow tables.
+- How to enable [incremental loading](../general-usage/incremental-loading) for efficient data extraction.
+- How to use build in ConnectionString credentials.
```
Comment (AstrakhantsevaAA) [inline docs/examples/connector_x_arrow/connector_x_arrow.py:20]: so we don't have Setup: Running this example on your machine section anymore?
do you think it's useless or it difficult to add in this implementation?

**Review by AstrakhantsevaAA — APPROVED** [review]

File: docs/examples/connector_x_arrow/connector_x_arrow.py (hunk 2 above)
Comment (sh-rp) [reply docs/examples/connector_x_arrow/connector_x_arrow.py:20]: it was a feedback from @burnash (at least if i recall correctly) to make these pages more concise. I can add this section back in though, i am not super opinionated on this.

**Pull request merged**